src/lambdas/auditMessageFirehoseReingest/deleteOrUpdateS3Objects.ts

```diff
@@ -0,0 +1,66 @@
+import { constants, gzipSync } from 'node:zlib'
+import { logger } from '../../services/logger'
+import { deleteS3Object } from '../../services/s3/deleteS3Object'
+import { putS3Object } from '../../services/s3/putS3Object'
+import { AuditEvent } from '../../types/auditEvent'
+import { S3ObjectDetails } from '../../types/s3ObjectDetails'
+import { objectToBase64 } from '../../utils/helpers/objectToBase64'
+
+export const deleteOrUpdateS3Objects = async (results: S3ObjectDetails[]) => {
+  return Promise.all(
+    results.map(async (result) => {
+      if (
+        result.auditEventsFailedReingest &&
+        result.auditEventsFailedReingest.length > 0
+      ) {
+        try {
```

Comment:
You could do with splitting some of this up into separate functions (within the same file, but just for readability). Having an `if` block followed by a `try/catch`, and then an `else` is a bit much and quite difficult to follow

Reply:
Have split up into functions now! Hope this is better?